app/Post_Types/Archive_Item/Rest.php

```diff
@@ -0,0 +1,343 @@
+use Diviner\Post_Types\Diviner_Field\Types\Text_Field;
+use Diviner\Post_Types\Diviner_Field\Types\Date_Field;
+use Diviner\Post_Types\Diviner_Field\Types\Taxonomy_Field;
+use Diviner\Post_Types\Diviner_Field\Types\CPT_Field;
+use Diviner\Post_Types\Diviner_Field\Types\Select_Field;
+use Diviner\Admin\Settings;
+use Diviner\CarbonFields\Helper;
+
+
+class Rest {
+
+	const FIELD_INDEX_ID = 'id';
+	const FIELD_INDEX_TITLE = 'title';
+	const FIELD_INDEX_TYPE = 'type';
+	const FIELD_INDEX_FIELD_ID = 'field_id';
+	const FIELD_INDEX_FIELD_POPUP = 'field_popup';
+
+	protected $container;
+	protected $fields;
+
+	public function init() {
+		$this->fields = $this->get_fields();
+	}
+
+	public function hooks() {
+		$action = 'after_setup_theme';
+		if ( DIVINER_IS_PLUGIN ) {
+			$action = 'plugins_loaded';
+		}
+		// add_filter( $action, [$this, 'init'], 10, 2 );
+		add_filter( 'rest_' . Archive_Item::NAME . '_query', [$this, 'rest_api_filter_add_vars'], 10, 2 );
+		add_action( 'rest_api_init', array( &$this,'custom_register_rest_fields') );
+	}
+
+	protected function get_fields() {
+		if (isset($this->fields)) {
+			return $this->fields;
+		}
+		$meta_query = array(
+			array(
+				'key'     => Helper::get_real_field_name(FieldPostMeta::FIELD_ACTIVE ),
+				'value'   => FieldPostMeta::FIELD_CHECKBOX_VALUE
+			),
+		);
+		$args = array(
+			'posts_per_page' => -1,
+			'fields' => 'ids',
+			'post_type' => Diviner_Field::NAME,
+			'meta_query' => $meta_query
+		);
+		$fields = [];
+		$posts_ids = get_posts($args);
+		foreach($posts_ids as $post_id) {
+			$fields[] = [
+				self::FIELD_INDEX_ID => $post_id,
+				self::FIELD_INDEX_TITLE => get_the_title($post_id),
+				self::FIELD_INDEX_TYPE => carbon_get_post_meta($post_id, FieldPostMeta::FIELD_TYPE, 'carbon_fields_container_field_variables'),
+				self::FIELD_INDEX_FIELD_ID => carbon_get_post_meta($post_id, FieldPostMeta::FIELD_ID, 'carbon_fields_container_field_variables'),
+				self::FIELD_INDEX_FIELD_POPUP => carbon_get_post_meta($post_id, FieldPostMeta::FIELD_BROWSE_DISPLAY, 'carbon_fields_container_field_variables'),
+			];
+		}
+		$this->fields = $fields;
+		return $this->fields;
+	}
+
+	public function decorate_taxonomy_args($field, $args, $request ) {
+		if (empty($request[$field[self::FIELD_INDEX_FIELD_ID]])) {
+			return $args;
+		}
+		return $args;
+	}
+
+
+	public function decorate_cpt_args($field, $args, $request ) {
+		if ( empty( $request[$field[self::FIELD_INDEX_FIELD_ID]] ) ) {
+			return $args;
+		}
+		$value = $request[$field[self::FIELD_INDEX_FIELD_ID]];
+
+		// error_log(print_r( $value, true ) , 3, "/Applications/MAMP/logs/php_error.log");
+
+		global $wp;
+		$vars = apply_filters( 'query_vars', $wp->public_query_vars );
+		if ( ! isset($args[ 'meta_query' ])) {
+			$args[ 'meta_query' ] = [
+				'relation'		=> 'AND',
+			];
+		}
+		$field_cpt_id = carbon_get_post_meta( $field[self::FIELD_INDEX_ID],FieldPostMeta::FIELD_CPT_ID );
```

Comment:
Make sure this is not empty.
Running this query against an empty field might lead to unexpected result and updating the wrong field.

Reply:
done